fdm-app/app/components/custom/soil/cards.tsx

```diff
@@ -145,19 +150,47 @@ export function SoilDataCards({
     )
     return (
         <div className="grid gap-4 grid-cols-1 sm:grid-cols-2 md:grid-cols-2 xl2:grid-cols-3 xl3:grid-cols-5">
-            {cards.map((card) => (
-                <SoilDataCard
-                    key={card.title}
-                    title={card.title}
-                    shortname={card.shortname}
-                    value={card.value}
-                    unit={card.unit}
-                    type={card.type}
-                    link={card.link}
-                    date={card.date}
-                    source={card.source}
-                />
-            ))}
+            {cards.map(
+                (card: {
+                    title: string
+                    shortname: string
+                    value: string | number | undefined
+                    label: string | undefined
+                    unit: string
+                    type: "numeric" | "enum"
+                    link: string
+                    date: Date
+                    source: string
+                    sourceLabel: string
+                }) => {
+                    if (card.value) {
+                        const sourceParam = soilParameterDescription.find(
+                            (x: { parameter: string }) => x.parameter === "a_source",
+                        )
+                        const sourceOption = sourceParam?.options?.find(
+                            (x: { value: string }) => x.value === card.source,
+                        )
+                        const sourceLabel =
+                            sourceOption?.label || card.source || "Onbekend"
+
+                        return (
+                            <SoilDataCard
+                                key={card.title}
```

Comment:
is key equal to title?

Reply:
Yes, I use title here,  as components are generated in a loop and need an unique identifier, and title is unique across the items